source/util.cpp

```diff
@@ -99,20 +100,35 @@ string indent(string const &code, string const &indentation)
 	return r;
 }
 
+std::string join( std::vector<std::string>& splits, std::string const& sep ) {
```

Comment:
`splits` should be passed by `const &`

Reply:
One thing I was not sure - did you leave prefix open such that you could actually add that as file name prefix? Eg if
prefix = /home/abc/foo
root_module=bar
Do you want the files to be created as 
/home/abc/foo/bar.cpp  or
/home/abc/foobar.cpp ?